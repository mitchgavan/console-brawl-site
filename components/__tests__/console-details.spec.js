import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConsoleDetails from '../console-list/console-details';

test('It renders correctly', () => {
  const tree = shallow(
    <ConsoleDetails
      key="playstation-1"
      name="Playstation 1"
      released="December 3, 1994"
      developer="Sony"
      description="Very cool."
      slug="playstation-1"
      reverseOrder={false}
      toggledOn="All"
    />,
  );
  expect(toJson(tree)).toMatchSnapshot();

  const treeReverseOrder = shallow(
    <ConsoleDetails
      key="playstation-1"
      name="Playstation 1"
      released="December 3, 1994"
      developer="Sony"
      description="Very cool."
      slug="playstation-1"
      reverseOrder
      toggledOn="All"
    />,
  );
  expect(toJson(treeReverseOrder)).toMatchSnapshot();

  const treeToggledOff = shallow(
    <ConsoleDetails
      key="playstation-1"
      name="Playstation 1"
      released="December 3, 1994"
      developer="Sony"
      description="Very cool."
      slug="playstation-1"
      reverseOrder={false}
      toggledOn="Nintendo"
    />,
  );
  expect(toJson(treeToggledOff)).toMatchSnapshot();
});
