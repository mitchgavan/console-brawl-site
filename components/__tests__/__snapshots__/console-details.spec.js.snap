// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div>
  <div
    className="root"
    data-jsx={2502043678}
  >
    <div
      className="row large-collapse"
      data-jsx={2502043678}
    >
      <div
        className="columns large-8 false"
        data-jsx={2502043678}
      >
        <img
          alt="Playstation 1"
          sizes="(min-width: 1024px) 800px, 100vw"
          src="/static/images/platforms/playstation-1@2x.jpg"
          srcSet="/static/images/platforms/playstation-1.jpg 400w,
               /static/images/platforms/playstation-1@2x.jpg 800w,
               /static/images/platforms/playstation-1@3x.jpg 1200w"
        />
      </div>
      <div
        className="columns large-4 false"
        data-jsx={2502043678}
      >
        <ConsoleDescription
          description="Very cool."
          developer="Sony"
          name="Playstation 1"
          released="December 3, 1994"
          reverseOrder={false}
          slug="playstation-1"
          toggledOn="All"
        />
      </div>
    </div>
    <_class
      css=".root[data-jsx=\\"2502043678\\"] {padding: 1rem 0;}@media screen and (min-width: 769px) {.root[data-jsx=\\"2502043678\\"] {padding: 3rem 0;}}"
      styleId={2502043678}
    />
  </div>
</div>
`;

exports[`It renders correctly 2`] = `
<div>
  <div
    className="root"
    data-jsx={2502043678}
  >
    <div
      className="row large-collapse"
      data-jsx={2502043678}
    >
      <div
        className="columns large-8 large-push-4"
        data-jsx={2502043678}
      >
        <img
          alt="Playstation 1"
          sizes="(min-width: 1024px) 800px, 100vw"
          src="/static/images/platforms/playstation-1@2x.jpg"
          srcSet="/static/images/platforms/playstation-1.jpg 400w,
               /static/images/platforms/playstation-1@2x.jpg 800w,
               /static/images/platforms/playstation-1@3x.jpg 1200w"
        />
      </div>
      <div
        className="columns large-4 large-pull-8"
        data-jsx={2502043678}
      >
        <ConsoleDescription
          description="Very cool."
          developer="Sony"
          name="Playstation 1"
          released="December 3, 1994"
          reverseOrder={true}
          slug="playstation-1"
          toggledOn="All"
        />
      </div>
    </div>
    <_class
      css=".root[data-jsx=\\"2502043678\\"] {padding: 1rem 0;}@media screen and (min-width: 769px) {.root[data-jsx=\\"2502043678\\"] {padding: 3rem 0;}}"
      styleId={2502043678}
    />
  </div>
</div>
`;

exports[`It renders correctly 3`] = `<div />`;
